### ISSUE TITLE
Typescript & Computed Property & VueTypes.arrayOf(VueTypes.shape())

### ISSUE THREAD

**Issue by uriannrima** [dwightjack/vue-types#36]
Hello,

Loving the library so far, finding it really useful, but I'm having  some problem while trying to create a computed property that needs to iterate on some property.

I have this scenario:

```
<script lang="ts">
import Vue from 'vue';
import VueTypes from 'vue-types';

export default Vue.extend({
  props: {
    classes: VueTypes.arrayOf(
      VueTypes.shape({
        name: VueTypes.string.isRequired,
        level: VueTypes.number.isRequired
      })
    ).isRequired
  },
  computed: {
    getSum(): number {
      return this.classes.reduce((acc, classe) => acc + classe.level, 0)
    }
  }
});
</script>
```
And it gives me this error, right above the reduce operation:

```
Operator '+' cannot be applied to types 'number' and 'VueTypeDef<number, number> & { required: true; }'.
(property) level: VueTypeDef<number, number> & {
    required: true;
}
```

Since "classes" is a array of VueTypeDef, I can't use it's properties (nor name or level) to compute something on it. If I swap "classe" to Any type, Typescript stops complaining, the code  works, but I'm trying to avoid the use of the Any type.

Another way that I found was to create a interface for my "shape":

```
interface Classe {
  name: string;
  level: number;
}
```
And "cast" VueTypes.arrayOf() to an PropOptions of it:

```
classes: VueTypes.arrayOf(
      VueTypes.shape({
        name: VueTypes.string.isRequired,
        level: VueTypes.number.isRequired
      })
    ).isRequired as PropOptions<Classe[]>
```

It worked, even though I found it a bit verbose.
Am I doing something wrong? Is there simpler way of using an arrayOf shapes?

Thanks again.

**Comment by dwightjack** [comment]
@uriannrima I think I've found the problem. 

I update the `shape` method signature from:

```ts
shape<S extends { [key: string]: VueProp | Prop<any> }>(obj: S): VueTypeShape<S>;
```
to:

```ts
shape<T>(obj: { [K in keyof T]?: Prop<T[K]> | VueProp<T[K]> }): VueTypeShape<T>;
```
With that signature everything should work fine even without using an interface.

I've published a new version under the beta tag. You can try it by running:

```
yarn add vue-types@beta
```

Let me know if it works for you.

**Comment by uriannrima** [comment]
Thank you for looking into it @dwightjack :D
I'll test it out and let you know about. (:

**Comment by dwightjack** [comment]
@uriannrima I just published the fix related to this issue and #43 as version `1.3.4`.

Closing this issue for now. Anyway feel free to reopen or open another issue for any further problem.
